src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -147,6 +153,8 @@ public ValueTask UnbindAsync(CancellationToken cancellationToken = default)
 
         public ValueTask DisposeAsync()
         {
+            _socketHandle?.Dispose();
```

Comment:
Would it be better to dispose the Socket first so it's not trying to utilize an already disposed handle?

Reply:
Done